### PR TITLE
[FINE] BZ#1499692-Added Loading indicator to Dialogs

### PR DESCRIPTION
**Pull request by chalettu**
This is a Fine specific PR for adding loading indicator.
@miq-bot add_label fine/yes
@miq-bot add_label enhancement
BZ https://bugzilla.redhat.com/show_bug.cgi?id=1499692
Original GH PR https://github.com/ManageIQ/manageiq-ui-service/pull/1258 for G release and master.

### CODE REVIEW

**Comment by AllenBW** [comment]
@chalettu can we get a gif of this in action?

File: client/app/states/catalogs/details/details.state.js [hunk 1]
```diff
@@ -13,64 +14,94 @@ function getStates() {
...
+        CollectionsApi.get('service_templates', serviceTemplateId, options).then((data) => {
+          resolve(data);
+        });
+      });
+      const allPromises = [dialogRequest, serviceTemplateRequest];
+      Promise.all(allPromises).then((data) => {
+        const SERVICE_TEMPLATE_RESPONSE = 1;
+        const DIALOGS_RESPONSE = 0;
+        dialogs = data[DIALOGS_RESPONSE];
+        vm.dialogs = dialogs.resources[0].content;
```
Comment (AllenBW) [inline client/app/states/catalogs/details/details.state.js]: screen shot 2017-12-14 at 7 55 07 am

Seeing this error when selecting a service, might want to test this value before assignment :-/

Then again maybe this is just an instance of a borked service catalog item 🤷‍♀️

**Comment by chalettu** [comment]
@AllenBW , here is the screenshot you requested.  Seems like your issue above might have been with that catalog item on your dev instance. 
sample_service

**Comment by chalettu** [comment]
Sample of dialogs with custom button details page. custom_button_details

**Pull request merged**